test/e2e/storage/persistent_volumes-local.go

```diff
@@ -806,26 +865,66 @@ func unmountTmpfsLocalVolume(config *localTestConfig, dir string, node *v1.Node)
+}
+
+func unmountBlockLocalVolume(config *localTestConfig, dir string, node *v1.Node) {
+	loopDev := getBlockLoopDev(dir, node)
+	By(fmt.Sprintf("Unmount block mount point %s on node %q at path %s/file", loopDev, node.Name, dir))
+	losetupDeleteCmd := fmt.Sprintf("sudo losetup -d %s", loopDev)
+	err := framework.IssueSSHCommand(losetupDeleteCmd, framework.TestContext.Provider, node)
+	Expect(err).NotTo(HaveOccurred())
+}
+
 // Create corresponding write and read commands
 // to be executed via SSH on the node with the local PV
-func createWriteAndReadCmds(testFileDir string, testFile string, writeTestFileContent string) (writeCmd string, readCmd string) {
-	writeCmd = createWriteCmd(testFileDir, testFile, writeTestFileContent)
-	readCmd = createReadCmd(testFileDir, testFile)
+func createWriteAndReadCmds(testFileDir string, testFile string, writeTestFileContent string, volumeType localVolumeType) (writeCmd string, readCmd string) {
+	writeCmd = createWriteCmd(testFileDir, testFile, writeTestFileContent, volumeType)
+	readCmd = createReadCmd(testFileDir, testFile, volumeType)
 	return writeCmd, readCmd
 }
 
-func createWriteCmd(testFileDir string, testFile string, writeTestFileContent string) string {
-	testFilePath := filepath.Join(testFileDir, testFile)
-	return fmt.Sprintf("mkdir -p %s; echo %s > %s", testFileDir, writeTestFileContent, testFilePath)
+func createWriteCmd(testDir string, testFile string, writeTestFileContent string, volumeType localVolumeType) string {
+	if volumeType == BlockLocalVolumeType {
+		// testDir is the block device.
+		testFileDir := filepath.Join("/tmp", testDir)
+		testFilePath := filepath.Join(testFileDir, testFile)
+		// Create a file containing the testFileContent.
+		writeTestFileCmd := fmt.Sprintf("mkdir -p %s; echo %s > %s", testFileDir, writeTestFileContent, testFilePath)
+		// sudo is needed when using ssh exec to node.
+		// sudo is not needed and does not exist in some containers (e.g. busybox), when using pod exec.
+		sudoCmd := fmt.Sprintf("SUDO_CMD=$(which sudo); echo ${SUDO_CMD}")
+		// Write the testFileContent into the block device.
+		writeBlockCmd := fmt.Sprintf("${SUDO_CMD} dd if=%s of=%s bs=512 count=100", testFilePath, testDir)
+		// Cleanup the file containing testFileContent.
+		deleteTestFileCmd := fmt.Sprintf("rm %s", testFilePath)
+		return fmt.Sprintf("%s && %s && %s && %s", writeTestFileCmd, sudoCmd, writeBlockCmd, deleteTestFileCmd)
+	} else {
+		testFilePath := filepath.Join(testDir, testFile)
+		return fmt.Sprintf("mkdir -p %s; echo %s > %s", testDir, writeTestFileContent, testFilePath)
+	}
 }
-func createReadCmd(testFileDir string, testFile string) string {
-	testFilePath := filepath.Join(testFileDir, testFile)
-	return fmt.Sprintf("cat %s", testFilePath)
+func createReadCmd(testFileDir string, testFile string, volumeType localVolumeType) string {
+	if volumeType == BlockLocalVolumeType {
+		// Read the beginning of the block device and print it in ascii.
```

Comment:
Might want to mention this in the function level comment. Not quite expecting that from the function.